lib/vmdb/settings.rb

```diff
@@ -198,5 +206,33 @@ def self.apply_settings_changes(resource, deltas)
       end
     end
     private_class_method :apply_settings_changes
+
+    def self.process_magic_value(resource, delta)
+      return false unless MAGIC_VALUES.include?(delta[:value])
+      case delta[:value]
+      when RESET_COMMAND
+        _log.info("resetting #{delta[:key]} on #{resource.class} id:#{resource.id}")
+        resource.settings_changes.where("key LIKE ?", "#{delta[:key]}%").destroy_all
+      when RESET_ALL_COMMAND
+        _log.info("resetting #{delta[:key]} to default for all resorces")
+        SettingsChange.where("key LIKE ?", "#{delta[:key]}%").destroy_all
+      end
+      true
+    end
+    private_class_method :process_magic_value
+
+    def self.walk_hash(hash, &block)
+      hash.each do |key, value|
```

Comment:
Prefer using the existing walk method, which already handles this.  Even so, I don't think Iike this manual thing at all, A reset is really just taking the parent's value at that level and replacing it.  Then, let the differ handle the rest.